src/pages/SignUpPage/SignUpPage.tsx

```diff
@@ -0,0 +1,342 @@
+import { Box, Typography, useTheme, Link } from '@mui/material';
+import React, { useState } from 'react';
+
+import FacebookStatic from './assets/Facebook.svg?url';
+import GithubStatic from './assets/Github.svg?url';
+import GoogleStatic from './assets/Google.svg?url';
+import {
+  getPageContainerStyles,
+  getFormStyles,
+  getLogoStyles,
+  getInputStyles,
+  getButtonStyles,
+  haveAccountTextStyles,
+  loginLinkStyles,
+  signupWithTextStyles,
+  getSocialIconStyles,
+  legalTextStyles,
+  legalLinkStyles,
+} from './SignUpPage.styles';
+
+import type { SignUpPageProps } from './SignUpPage.types';
+
+const FacebookIcon: string | undefined = FacebookStatic ?? undefined;
+const GithubIcon: string | undefined = GithubStatic ?? undefined;
+const GoogleIcon: string | undefined = GoogleStatic ?? undefined;
+
+type LogoFallbackProps = {
+  orientation?: string;
+  width?: string | number;
+  height?: string | number;
+  alt?: string;
+};
+
+type FormInputFallbackProps = {
+  label?: string;
+  value?: string;
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  placeholder?: string;
+  errorMessage?: string;
+  sx?: Record<string, unknown>;
+  type?: string;
+};
+
+type PrimaryButtonFallbackProps = {
+  children?: React.ReactNode;
+  onClick?: (e: React.FormEvent) => void;
+  disabled?: boolean;
+  loading?: boolean;
+  sx?: Record<string, unknown>;
+};
+
+const Logo: React.ComponentType<LogoFallbackProps> = () => <div data-testid="logo-fallback" />;
+
+const FormInput: React.FC<FormInputFallbackProps> = (props) => (
+  <>
+    <Box
+      data-testid={`form-input-fallback-${props.placeholder ?? 'input'}`}
+      component="div"
+      sx={(props.sx as unknown as Record<string, unknown>) ?? { width: '100%', height: '40px' }}
```

Comment:
Default sx fallback in the FormInput component uses hardcoded px: { width: '100%', height: '40px' }. Project guideline forbids hardcoded px. Prefer theme.spacing() or rems via theme.typography.

**Sugerencia:**
```typescript
Use a theme-based size: e.g. { width: '100%', height: theme.spacing(5) } (and type sx prop correctly so theme is available).
```